Move package branch dialog to dedicated view

krauselukas opened the pull request:
The package branch dialog deserves it's own view, since
there are to quite some options which lead to a pretty
packed and unclear modal.

<!---
If you haven't done so already, please read the CONTRIBUTING.md file to learn
how we work and what we expect from all contributors.

https://github.com/openSUSE/open-build-service/blob/master/CONTRIBUTING.md

In order to make it as easy as possible for other developers to review your
pull request we ask you to:

- Explain what this PR is about in the description
- Explain the steps the reviewer has to follow to verify your change
- If the reviewer needs sample data to verify your change, please explain how to
  create that data
- If you include visual changes in this PR, please add screenshots or GIFs
- If you address performance in this PR, add benchmark data or explain how the
  reviewer can benchmark this

This is a good PR description example:

Hey Friends,

this introduces labels for the different build result states on the project
monitor page. This makes it easier to get a visual overview of what is going on
in your project.

To verify this feature

- Enable the interconnect to build.opensuse.org
- Create the project home:Admin
- Add 'openSUSE Tumbleweed' as a repository to the project
- Branch a couple of packages into the project:
  ```
  for i in `osc -A http://0.0.0.0:3000 ls openSUSE.org:home:hennevogel`; do osc -A http://0.0.0.0:3000 copypac openSUSE.org:home:hennevogel $i home:Admin; done
  ```
- Visit the monitor page and see the new labels for the different states.

Here is a screenshot of how it looks:

** Before **
![Screenshot of the project monitor](https://example.com/screenshot1.png)

** After **
![Screenshot of the project monitor](https://example.com/screenshot2.png)

-->

krauselukas commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[M] Review Modals for Package#show](https://trello.com/c/bMLbySQv/1401-m-review-modals-for-packageshow)

dmarcoux commented on src/api/app/controllers/webui/package_controller.rb:
```diff
@@ -373,6 +373,11 @@ def branch_diff_info
     }
   end
 
+  def branch_dialog
```
We don't have a modal/dialog anymore, so this action name is misleading. I suggest naming this action `branch` and renaming the `branch` action below to `create_branch` (maybe we can find a better name). Then we need to adapt the routes a bit.

We can either have one route `branch` with the `GET` going to `package#branch` and the `POST` going to `package#create_branch`. This is what Rails does for `index` and `create`. They both point to the same route, but depending on the HTTP verb, the action varies.

The other option is to have a separate route, so two routes just like the actions, so `GET branch` and `POST create_branch`.

damianatorrpm commented:
If it is on it's own page maybe 'more options' should be expanded.

krauselukas commented:
@hennevogel I brought back the ```More options``` area, I agree that this brings in unnecessary complexity that many people don't need. I reworked a bit the hierarchy on it, added some padding and removed the strong tags from the form labels.
@damianatorrpm moving the things to the right side doesn't really bring an advantage from my point of view. On desktop sized screens we have both a lot of space on the right and on the bottom.

krauselukas commented:
I would ignore codeclimate, the problems are not introduced by this PR (only gets triggered because I renamed a method) and fixing it is a big refactor.

hennevogel commented on src/api/app/views/webui/package/branch.html.haml:
```diff
@@ -0,0 +1,28 @@
+:ruby
+  @pagetitle = "Branch #{@package}"
+
+.card.mb-3
```
Inside containers you normally use [the grid](https://getbootstrap.com/docs/4.4/layout/grid/) to place things, not cards.

krauselukas replied:
@hennevogel I applied the grid system, but I kept the card. We use them almost everywhere. Not sure if I understood correctly, do you want to get rid of the way we use the cards to structure the view or was it more about the missing grid?

hennevogel replied:
You now have a wild mixture of card and grid. The grid is for the layout, the card is styling the content. Build a grid, fill it with content.

vpereira replied:
@hennevogel I see that in many PRs we are going with `card.mb-3` and you are against it. Could you please explain us how you want it to avoid everyone doing their own interpretation of what you are saying?

hennevogel replied:
we talked about this offline. grid first, then put content onto it is our new motto! :smile:

hennevogel commented on src/api/app/views/webui/package/branch.html.haml:
```diff
@@ -0,0 +1,28 @@
+          .p-2
+            = hidden_field_tag :linked_project, @project.name
+            = hidden_field_tag :linked_package, @package.name
+            = render partial: 'webui/shared/package_branch_form', locals: { show_project_field: true,
```
Using the grid also will make it possible to avoid overly long input fields.

krauselukas replied:
It's fixed, the applied grid keeps the input fields in place :)

dmarcoux commented:
@hennevogel Please have another look if changes are still needed.
